### ISSUE TITLE
Unhelpful error message (`Error: unexpected token: "("`) when defining method that starts with a capital letter.

### ISSUE THREAD

**Issue by Zopolis4** [crystal-lang/crystal#15316]
## Bug Report

The following code fails to compile:
```crystal
def Foo(bar)
 puts "qux #{bar}"
end

Foo("quux")
```
with the following error:
```
In temp.cr:1:8

 1 | def Foo(bar)
            ^
Error: unexpected token: "("
```

I believe the starting with the capital letter is the issue, because renaming the method to `foo` or `fFoo` or `_Foo` fixed it (successfully printing the output of `qux quux`), but the error message does not make that clear. 

I was additionally unable to find any documentation detailing that method names couldn't start with capital letters, so assuming I am not incorrect about the cause of the error it may help to update the relevant documentation to take note of that in addition to updating the error message. 

Tested on:
```
Crystal 1.14.0 [dacd97bcc] (2024-10-09)

LLVM: 18.1.6
Default target: x86_64-unknown-linux-gnu
```

**Comment by devnote-dev** [comment]
You're on the right track — Crystal allows `def Foo.bar` syntax to define a method on a type without explicitly reopening it, I think in this case it's expecting the remaining part (`.bar`), when it should rightfully error on "Foo".

**Comment by Zopolis4** [comment]
Is that also the case for `def FOO.bar`?

Because I get the same when doing:
```crystal
def FOO(bar)
 puts "qux #{bar}"
end

FOO("quux")
```
which then gives
```
In EEEE.cr:1:8

 1 | def FOO(bar)
            ^
Error: unexpected token: "("
```

**Comment by devnote-dev** [comment]
Yes, any identifier that starts with a capital letter is initially parsed as a constant, which usually refers to a type.

**Comment by Blacksmoke16** [comment]
This is called out in https://crystal-lang.org/reference/1.14/syntax_and_semantics/methods_and_instance_variables.html:

> Method names begin with a lowercase letter and, as a convention, only use lowercase letters, underscores and numbers.

**Comment by Zopolis4** [comment]
Got it, thanks. In that case, its only the error message that needs to be updated to point to the actual problem.
